pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -708,12 +708,12 @@ private static Expression compileFunctionExpression(SqlBasicCall functionNode) {
       case OTHER:
       case OTHER_FUNCTION:
       case DOT:
-        functionName = functionNode.getOperator().getName().toUpperCase();
+        functionName = canonicalize(functionNode.getOperator().getName());
         if (functionName.equals("ITEM") || functionName.equals("DOT")) {
           // Calcite parses path expression such as "data[0][1].a.b[0]" into a chain of ITEM and/or DOT
           // functions. Collapse this chain into an identifier.
           StringBuffer path = new StringBuffer();
-          compilePathExpression(functionName, functionNode, path);
```

Comment:
Can you get rid of this StringBuffer while you’re here?

Reply:
Do you mean replacing it with `StringBuilder`? Done

Reply:
👍